cli/mk_data/README.md

```diff
@@ -0,0 +1,96 @@
+# mk_data
+
+This program generates test data used for mocking during unit tests.
+Cases are specified in a config file.
+
+```
+Generate mock test data from a config file
+
+Usage: mk_data [OPTIONS] <PATH>
+
+Arguments:
+  <PATH>  The path to the config file
+
+Options:
+  -f, --force            Regenerate all data and overwrite existing data
+  -o, --output <OUTPUT>  The path to the directory in which to store the output [default: $PWD/generated]
+  -i, --input <INPUT>    The path to the directory containing the input data [default: $PWD/input_data]
+  -q, --quiet            Don't show a spinner
+  -h, --help             Print help
+```
+
+## Config file
+
+The `[vars]` section is special in that it doesn't specify jobs to run or
+generate output.
+
+The directory structure generated is as follows:
+```
+<root>/
+    init/
+        <job_name>.json
+    resolve/
+        <job_name>.json
+    search/
+        <job_name>.json
+    show/
+        <job_name>.json
+    envs/
+        <job_name>/
+            <job_name.json>
+            manifest.toml
+            manifest.lock
+```
+
+The only real exception here is that the directories in the `envs` section
+are created manually using the job config.
+
+### Job config
+
+Each job is executed in a temporary directory inside the output directory.
+
+The config for a job is as follows:
+```
+files    := [str] | null
+pre_cmd  := str | null
+cmd      := str | null
+post_cmd := str | null
+```
+
+Semantics:
+- `files` are relative paths specified relative to the input data directory.
+- `files` are copied from the input data directory into the working directory before anything else happens.
+- `pre_cmd` executes next, but no response file is generated during execution.
+- `cmd` is executed next and a response file is generated.
```

Comment:
nonblocking question: is there a difference between `pre_cmd = cmd1, cmd = cmd2` and `cmd = cmd1; cmd2`?

Reply:
Strictly as written, your second case would merge responses from cmd1 and cmd2 since the default behavior is to merge rather than overwrite (last time I checked).

The other difference is that the dump environment variable isn't set for pre_cmd.